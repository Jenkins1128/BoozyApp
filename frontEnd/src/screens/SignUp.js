import React from 'react';
import {Animated, StyleSheet, Text, View, TextInput,  TouchableOpacity, ImageBackground } from 'react-native';

var isHidden = true;
export default class SignUp extends React.Component{
    
    constructor(props){
        super(props);
        this.state = {
            email: '',
            password: '',
            bounceValue: new Animated.Value(50)
        } 
    }
    
    render(){
        return (    
                <View style={styles.container}>      
                        <ImageBackground source={require('./background.jpg')} style={styles.backgroundImage} >
                        </ImageBackground>
                        <Text style={styles.BoozyText}>Boozy</Text>
                        <Text style={styles.BoozySlogan}>Happy hours, Happy you.</Text>
        
                        <TextInput
                            style={styles.emailInput}
                            onChangeText={(email) => this.setState({email})}
                            value={this.state.email}
                            placeholder='Email'
                            placeholderTextColor='#EB8873'
                            underlineColorAndroid='transparent'>
                        </TextInput> 
                        <TextInput
                            style={styles.passwordInput}
                            secureTextEntry={true}
                            onChangeText={(password) => this.setState({password})}
                            value={this.state.password}
                            placeholder='Password'
                            placeholderTextColor='#EB8873'
                            underlineColorAndroid='transparent'>
                        </TextInput>     
                        <TouchableOpacity onPress={this.login.bind(this)} style={styles.loginButton}>      
                            <Text style={styles.logInText}>Log In</Text>
                        </TouchableOpacity>
                        <TouchableOpacity onPress={this.signup.bind(this)} style={styles.signUpButton}>
                
                            <Text style={styles.signUpText}>Sign Up</Text>
                        </TouchableOpacity> 
                        <Animated.View style={[styles.subView,{transform: [{translateY: this.state.bounceValue}]}]} >
                            <Text style={{fontSize: 20, fontWeight: 'bold', color:"red"}}>Username already exists.</Text>
                        </Animated.View>     
                </View>
        );
    }

    signup(){
        const axios = require('axios').default;
        if(this.state.email.trim().length > 0 && this.state.password.trim().length > 0){
        axios.post(`https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/register`, {
                email: this.state.email,
                password: this.state.password
              }, {
                headers: {
                    'Accept': 'application/json, text/plain, */*',
                    'Content-Type': 'application/json'
                }
              }).then((response) => {
                console.log("Registered!")
                //this.props.navigation.navigate('LoginPage');
                //console.log(response);
                this.login();
              })
              .catch((error) => {
                console.log(error);
                if(isHidden){
                    this.showErrorOverlay();
                }
                
              });

        }   
    }

    showErrorOverlay(){
        var toValue = 400;
        if(isHidden) {
          toValue = -150;
        }
        Animated.spring(
          this.state.bounceValue,
          {
            toValue: toValue,
            velocity: 5,
            tension: 2,
            friction: 8,
          }
        ).start();
        isHidden = !isHidden;
    }

    login(){
        if(!isHidden){
            this.showErrorOverlay();
        }
        this.props.navigation.navigate('Login');
    }

    goToHome(){
        this.props.navigation.navigate('Home');
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  header: {
        backgroundColor: '#E91E63',
        justifyContent: 'center',
        alignItems: 'center',
        borderBottomWidth: 10,
        borderBottomColor: '#ddd'
    },
    headerText: {
        backgroundColor: '#fc0',
        width: '100%', 
        height: '100%' 
    },
    backgroundImage: {
        flex: 1,
        width: '100%',
        height: '100%',
        justifyContent: "center",
        alignItems: "center",
        opacity: 0.95
    },
    subView: {
        position: "absolute",
  
        bottom: 0,
        left: 0,
        right: 0,
        backgroundColor: "#FFFFFF",
        height: 30,
        alignItems: 'center',
    },
    BoozyText:{
        color: '#EB8873',
        fontSize: 90,
        fontFamily: 'Bradley Hand',
        position: 'absolute',
        top: 125,
        fontWeight: 'bold',
        alignSelf: 'center',
    },
    BoozySlogan:{
        color: 'white',
        fontSize: 25,
        fontFamily: 'Bradley Hand',
        position: 'absolute',
        top: 340,
        fontWeight: 'bold',
        alignSelf: 'center',
    },
    scrollContainer: {
        flex: 1,
        marginBottom: 100
    },
    footer: {
        position: 'absolute',
        bottom: 0,
        left: 0,
        right: 0,
        zIndex: 10
    },
    usernameInput: {
        position: 'absolute',
        zIndex: 11,
        top: 400,
        width: 280,
        borderRadius: 30,
        alignSelf: 'center',
        color: '#FFFFFF',
        padding: 30,
        backgroundColor: '#FFFFFF',
        borderTopWidth: 2,
        borderTopColor: '#ededed',
    },
    emailInput: {
        position: 'absolute',
        zIndex: 11,
        top: 400,
        width: 280,
        borderRadius: 30,
        alignSelf: 'center',
        color: '#EB8873',
        padding: 30,
        backgroundColor: '#FFFFFF',
        borderTopWidth: 2,
        borderTopColor: '#ededed',
    },
    passwordInput: {
        position: 'absolute',
        zIndex: 11,
        top: 490,
        width: 280,
        borderRadius: 30,
        alignSelf: 'center',
        color: '#EB8873',
        padding: 30,
        backgroundColor: '#FFFFFF',
    },
    signUpButton: {
        position: 'absolute',
        zIndex: 14,
        right: 17,
        bottom: 30,
        width: 380,
        height: 90,
        borderRadius: 70,
        alignItems: 'center',
        justifyContent: 'center',
        elevation: 8,
        color: '#fff',
        padding: 20,
        backgroundColor: '#EB8873',
    },
    loginButton: {
        position: 'absolute',
        zIndex: 14,
        right: 10,
        top: 50,
        width: 100,
        height: 60,
        borderRadius: 70,
        alignItems: 'center',
        justifyContent: 'center',
        elevation: 8,
        color: '#fff',
        padding: 20,
        backgroundColor: '#EB8873',
    },
    logInText: {
        color: 'white',
        fontSize: 14,
        fontWeight: 'bold'
    },
    signUpText: {
        color: 'white',
        fontSize: 24,
        fontWeight: 'bold'
    }
});
