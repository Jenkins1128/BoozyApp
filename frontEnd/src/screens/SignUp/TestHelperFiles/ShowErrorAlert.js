import { Alert } from 'react-native';

const ShowErrorAlert = {
	showErrorAlert: (errorString) => {
		Alert.alert('Uh oh', errorString, [{ text: 'OK' }]);
	}
};

export default ShowErrorAlert;
