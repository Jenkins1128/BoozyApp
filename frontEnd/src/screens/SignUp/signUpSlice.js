import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const signUpAsync = createAsyncThunk('signUpAsync/status', async (data, { rejectWithValue }) => {
	console.log(data);
	try {
		const response = await axios.post(data.url, {
			email: data.email,
			password: data.password
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const signUpSlice = createSlice({
	name: 'signUpAsync',
	initialState,
	reducers: {
		updateEmail: (state, { payload }) => {
			state.currentState = { ...state.currentState, email: payload.email };
		},
		updatePassword: (state, { payload }) => {
			state.currentState = { ...state.currentState, password: payload.password };
		},
		reset: (state) => {
			state.currentState = {};
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(signUpAsync.pending, (state) => {
				state.currentState = { ...state.currentState, requestStatus: 'pending', isLoading: true };
			})
			.addCase(signUpAsync.fulfilled, (state) => {
				state.currentState = { ...state.currentState, requestStatus: 'fulfilled', isLoading: false };
			})
			.addCase(signUpAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, requestStatus: 'rejected', isLoading: false };
			});
	}
});

export const { updateEmail, updatePassword, reset } = signUpSlice.actions;
export const selectSignUpState = (state) => state.signup.currentState;

export default signUpSlice.reducer;
