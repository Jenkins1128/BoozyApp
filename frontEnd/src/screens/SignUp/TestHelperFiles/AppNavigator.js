import 'react-native-gesture-handler';
import * as React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import SignUp from '../SignUp';
import LogIn from '../../LogIn/LogIn';

const { Screen, Navigator } = createStackNavigator();

const Navigation = () => {
	return (
		<Navigator>
			<Screen name='SignUp' component={SignUp} />
			<Screen name='Login' component={LogIn} />
		</Navigator>
	);
};
export default Navigation;
