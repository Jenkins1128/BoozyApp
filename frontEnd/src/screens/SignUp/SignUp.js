import React, { useEffect } from 'react';
import { Alert, View, StyleSheet, ImageBackground, TouchableWithoutFeedback, Keyboard, SafeAreaViewComponent, SafeAreaView } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { updateEmail, updatePassword, resetStatus, signUpAsync, selectSignUpState } from './redux/signUpSlice';
import background from '../../images/background.jpg';
import SignUpHeader from './SignUpHeader/SignUpHeader';
import SignUpInput from './SignUpInput/SignUpInput';

const SignUp = ({ navigation }) => {
	const state = useSelector(selectSignUpState);
	const dispatch = useDispatch();

	useEffect(() => {
		if (isEmpty(state)) {
			return;
		}
		switch (state.requestStatus) {
			case 'idle':
				return;
			case 'rejected':
				showErrorAlert('Username already exists...');
				break;
			case 'fulfilled':
				goToLogin();
				break;
		}
		dispatch(resetStatus());
	}, [state]);

	const isEmpty = (currentState) => {
		for (const x in currentState) {
			return false;
		}
		return true;
	};

	const showErrorAlert = (errorString) => {
		Alert.alert('Uh oh', errorString, [{ text: 'OK' }]);
	};

	const goToLogin = () => {
		navigation.navigate('Login');
	};

	const signupPressed = () => {
		if (isEmpty(state) || !state.email.trim().length || !state.password.trim().length) {
			showErrorAlert('Please fill out all fields.');
			return;
		}
		if (state.password.trim().length < 6) {
			showErrorAlert('Password needs to be greater than 5 characters.');
			return;
		}
		dispatch(signUpAsync({ email: state.email, password: state.password }));
	};

	return (
		<TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
			<View style={styles.container}>
				<ImageBackground source={background} style={styles.backgroundImage}>
					<SignUpHeader state={state} goToLogin={goToLogin} />
					<SignUpInput state={state} dispatch={dispatch} updateEmail={updateEmail} updatePassword={updatePassword} signupPressed={signupPressed} />
				</ImageBackground>
			</View>
		</TouchableWithoutFeedback>
	);
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center'
	},
	backgroundImage: {
		flex: 1,
		width: '100%',
		height: '100%',
		opacity: 0.95
	}
});

export default SignUp;
