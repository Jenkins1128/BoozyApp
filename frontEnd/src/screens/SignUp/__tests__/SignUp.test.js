import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import SignUp from '../SignUp';
import { render, cleanup, fireEvent } from '@testing-library/react-native';
import isEmpty from '../TestHelperFiles/IsEmpty';
import ShowErrorAlert from '../TestHelperFiles/ShowErrorAlert';
import { NavigationContainer } from '@react-navigation/native';
import AppNavigator from '../TestHelperFiles/AppNavigator';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('<SignUp />', () => {
	const mockStore = configureStore([]);
	const signupReducer = {
		signup: {
			currentState: {
				requestStatus: 'idle',
				isLoading: false,
				email: 'xxx@gmail.com',
				password: 'xxx'
			}
		},
		login: {
			currentState: {
				requestStatus: 'idle',
				isLoading: false,
				email: 'xxx@gmail.com',
				password: 'xxx'
			}
		}
	};

	afterEach(cleanup);

	//snapshot
	it('should match snapshot', () => {
		const store = mockStore(signupReducer);
		const signup = (
			<Provider store={store}>
				<SignUp />
			</Provider>
		);
		const tree = render(signup).toJSON();
		expect(tree).toMatchSnapshot();
	});

	//content
	it('should wrap view with a flexible wrapper', () => {
		const store = mockStore(signupReducer);
		const signup = (
			<Provider store={store}>
				<SignUp />
			</Provider>
		);
		const rendered = render(signup);
		const keyboardAvoidingViewComponent = rendered.getByTestId('container');
		const given = keyboardAvoidingViewComponent.props.style;
		//padding-bottom 0 added from KeyboardAvoidingView behavior prop
		const result = [{ flex: 1 }, { paddingBottom: 0 }];
		expect(result).toMatchObject(given);
	});

	it('should wrap backgroundImage with a flexible wrapper and centers items', () => {
		const store = mockStore(signupReducer);
		const signup = (
			<Provider store={store}>
				<SignUp />
			</Provider>
		);
		const rendered = render(signup);
		const innerViewComponent = rendered.getByTestId('inner');
		const given = innerViewComponent.props.style;
		const result = {
			flex: 1,
			justifyContent: 'center',
			alignItems: 'center'
		};
		expect(given).toMatchObject(result);
	});

	it('should wrap a flexible wrapper around the Image Background component, width/height = 100%, and opacity = 0.95', () => {
		const store = mockStore(signupReducer);
		const signup = (
			<Provider store={store}>
				<SignUp />
			</Provider>
		);
		const rendered = render(signup);
		const imageBackgroundComponent = rendered.getByTestId('backgroundImage');
		const given = imageBackgroundComponent.props.style;
		const result = [{ bottom: 0, left: 0, position: 'absolute', right: 0, top: 0 }, { height: '100%', width: '100%' }, undefined];
		expect(given).toMatchObject(result);
	});

	//functions
	describe('isEmpty', () => {
		it('should return true', () => {
			const currentState = {};
			const result = isEmpty(currentState);
			expect(result).toBeTruthy();
		});
	});

	describe('showErrorAlert', () => {
		it('calls sideEffect', () => {
			jest.spyOn(ShowErrorAlert, 'showErrorAlert');
			ShowErrorAlert.showErrorAlert('Username already exists...');
			expect(ShowErrorAlert.showErrorAlert).toHaveBeenCalledTimes(1);
			expect(ShowErrorAlert.showErrorAlert).toHaveBeenCalledWith('Username already exists...');
		});
	});

	describe('goToLogin', () => {
		it('calls sideEffect', () => {
			const store = mockStore(signupReducer);
			const component = (
				<Provider store={store}>
					<NavigationContainer>
						<AppNavigator />
					</NavigationContainer>
				</Provider>
			);

			const { getByTestId } = render(component);
			const toClick = getByTestId('goToLogin');

			fireEvent(toClick, 'press');
			const loginButton = getByTestId('loginButton');

			expect(loginButton).toBeTruthy();
		});
	});
});
