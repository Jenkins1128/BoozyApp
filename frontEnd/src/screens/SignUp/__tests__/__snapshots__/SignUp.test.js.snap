// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> should match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="container"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="inner"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
          "height": "100%",
          "opacity": 0.95,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/background.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
        testID="backgroundImage"
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "marginTop": 40,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-end",
              "backgroundColor": "#EB8873",
              "borderRadius": 70,
              "color": "#fff",
              "height": 60,
              "justifyContent": "center",
              "marginRight": 5,
              "opacity": 1,
              "padding": 20,
              "width": 100,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Log In
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#EB8873",
              "fontFamily": "BradleyHandBold",
              "fontSize": 90,
            }
          }
        >
          Boozy
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "BradleyHandBold",
              "fontSize": 25,
            }
          }
        >
          Happy hours, Happy you.
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 3,
            "paddingHorizontal": 40,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="#EB8873"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 30,
              "borderTopColor": "#ededed",
              "borderTopWidth": 2,
              "color": "#EB8873",
              "marginBottom": 10,
              "padding": 30,
            }
          }
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="#EB8873"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 30,
              "color": "#EB8873",
              "marginBottom": 50,
              "padding": 30,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#EB8873",
              "borderRadius": 70,
              "color": "#fff",
              "elevation": 8,
              "height": 90,
              "justifyContent": "center",
              "opacity": 1,
              "width": "80%",
            }
          }
          testID="signupButton"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
